### ISSUE TITLE
TS Spec doesn't match implementation re method signature definitions

### ISSUE THREAD

**Issue by JacksonKearl** [microsoft/TypeScript#36680]
[TS Spec § 3.9.5](https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#395-method-signatures) says:

> A method signature of the form
>```ts 
> f < T1, T2, ... > ( p1, p2, ... ) : R
>```
>is equivalent to the property declaration
>```ts
>f : { < T1, T2, ... > ( p1, p2, ... ) : R }
>```

But this is not the case with `--strictFunctionTypes`:
```ts
interface IFoo1 {
    bar(baz?: number): void
}
class Foo1 implements IFoo1 {
    bar: (baz: number) => void = (n) => void 0;
}

interface IFoo2 {
    bar: { (baz?: number): void }
}
class Foo2 implements IFoo2 {
    bar: (baz: number) => void = (n) => void 0; // err: Property 'bar' in type 'Foo2' is not assignable to the same property in base type 'IFoo2'.
}
```

I understand that the method bivarience in general is intentional as per the [PR](https://github.com/Microsoft/TypeScript/pull/18654):
>With this PR we introduce a --strictFunctionTypes mode in which function type parameter positions are checked contravariantly instead of bivariantly. *The stricter checking applies to all function types, except those originating in method or construcor declarations*

But it would be nice if the spec didn't state the syntaxes were equivalent when they aren't.

**Comment by weswigham** [comment]
The spec is for TS 1.8. We are on version 3.8. Needless to say, the spec is out of date. Do note, the [large and still incomplete](https://github.com/microsoft/TypeScript/issues?q=is%3Aopen+is%3Aissue+label%3ASpec) list of things in the spec which need to be updated. I would not read the TS spec to learn TS.

**Comment by JacksonKearl** [comment]
Does the TS team have a recommended way to learn TS? The docs don't mention any of this.

**Comment by weswigham** [comment]
The [handbook](http://www.typescriptlang.org/docs/handbook/basic-types.html) is the place to go.

**Comment by JacksonKearl** [comment]
http://www.typescriptlang.org/docs/handbook/type-compatibility.html#function-parameter-bivariance

This doesn't mention the behaviour either. Could this be modified instead of the spec?

**Comment by weswigham** [comment]
Yes, it can - we take handbook suggestions/PRs [over here](https://github.com/microsoft/TypeScript-Handbook)! However we're also [building the content for a fancy new handbook right now](https://microsoft.github.io/TypeScript-New-Handbook/outline/) (which we'll also take content [suggestions on/for](https://github.com/microsoft/TypeScript-New-Handbook)).
